# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/nsBaseWidget.cpp
# Commit: 45d5b3283413
# Full Hash: 45d5b32834133797711925dba48f55d94fa640f1
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2019-07-30 21:50:30
# Description:
#   Bug 1569565 - Fix error handling in nsBaseWidget::CreateCompositorSession() r=nical
#   
#   When ClientLayerManager::Initialize() fails, we also need to clear LayerManager.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D39817
# ==============================================================================

diff -r 64200237b839 -r 45d5b3283413 widget/nsBaseWidget.cpp
--- a/widget/nsBaseWidget.cpp	Tue Jul 30 12:04:25 2019 +0000
+++ b/widget/nsBaseWidget.cpp	Tue Jul 30 08:19:04 2019 +0000
@@ -1316,6 +1316,7 @@
       if (textureFactoryIdentifier.mParentBackend ==
           LayersBackend::LAYERS_NONE) {
         DestroyCompositor();
+        lm = nullptr;
       }
     }
 
